src/sentry/utils/batching_kafka_consumer.py

```diff
@@ -168,6 +169,41 @@ def __record_timing(self, metric, value, tags=None):
         sample_rate = self.__metrics_sample_rates.get(metric, settings.SENTRY_METRICS_SAMPLE_RATE)
         return self.__metrics.timing(metric, value, tags=tags, sample_rate=sample_rate)
 
+    def _wait_for_topics(self, admin_client, topics, timeout=10):
+        """
+        Make sure that the provided topics exist and have non-zero partitions in them.
+        """
+        for topic in topics:
```

Comment:
I know this is only 3 topics right now but I cannot help but feeling we should be using a thread pool or something here.

Reply:
Yes, but as you mentioned, the added complexity would probably not worth it right now.